priv/features/contexts/payment_context.exs

```diff
@@ -0,0 +1,129 @@
+defmodule PaymentContext do
+  @moduledoc """
+  Provides step definitions for the expected, happy cases
+  """
+
+  use WhiteBread.Context
+  require Logger
+
+  alias Itest.Transactions.Currency
+  alias Itest.Account
+  alias Itest.Client
+  alias Itest.StandardExitClient
+  alias Itest.InFlightExitClient
+
+  @finality_margin_by_blocks 12
+
+  @default_timeout 30_000
+
+  scenario_timeouts(fn _feature, scenario ->
+    case scenario.name do
+      "Alice sends Bob funds" -> @default_timeout * 2
+      "Alice starts a Standard Exit" -> @default_timeout * 10
+      "Alice starts an In Flight Exit" -> @default_timeout * 30
+      _ -> @default_timeout
+    end
+  end)
+
+  scenario_starting_state(fn _ ->
+    {:ok, [{alice, alice_pkey}, {bob, bob_pkey}]} = Account.take_accounts(2)
+
+    %{
+      alice: %{address: alice, key: alice_pkey},
+      bob: %{address: bob, key: bob_pkey}
+    }
+  end)
+
+  when_(~r/^(?<owner>\w+) deposits "(?<amount>[^"]+)" ETH$/, &deposits/2)
+
+  when_(~r/^(?<sender>\w+) sends (?<receiver>\w+) "(?<amount>[^"]+)" ETH$/, &sends/2)
+
+  when_(~r/^(?<owner>\w+) starts a standard exit$/, &starts_standard_exit/2)
+
+  when_(~r/^(?<owner>\w+) starts an in flight exit$/, &starts_in_flight_exit/2)
+
+  then_(~r/^(?<owner>\w+) should have "(?<amount>[^"]+)" ETH$/, &balance_for/2)
+
+  then_(~r/^(?<owner>\w+) should have "(?<amount>[^"]+)" ETH after finality margin$/, &balance_after_finality_for/2)
+
+  def deposits(%{} = state, %{owner: owner, amount: amount}) do
+    %{address: address} = get_account(owner, state)
+
+    {:ok, _receipt_hash} =
+      amount
+      |> Currency.to_wei()
+      |> Client.deposit(address, Account.vault())
+
+    {:ok, state}
+  end
+
+  def sends(%{} = state, %{sender: sender, receiver: receiver, amount: amount}) do
+    %{address: sender_address, key: sender_key} = get_account(sender, state)
+    %{address: receiver_address} = get_account(receiver, state)
+
+    {:ok, [sign_hash, typed_data, _txbytes]} =
+      Client.create_transaction(Currency.to_wei(amount), sender_address, receiver_address)
+
+    _ = Client.submit_transaction(typed_data, sign_hash, sender_key)
+
+    {:ok, state}
+  end
+
+  def balance_for(%{} = state, %{owner: owner, amount: amount}) do
+    %{address: address} = get_account(owner, state)
+
+    expecting_amount = Currency.to_wei(amount)
+    balance = Client.get_balance(address, expecting_amount)
+    balance = if balance == [], do: 0, else: balance["amount"]
+
+    assert expecting_amount == balance, "Expecting #{owner} balance to be #{expecting_amount}, was #{balance}"
+
+    {:ok, state}
+  end
+
+  def balance_after_finality_for(state, arguments) do
+    Process.sleep(@finality_margin_by_blocks * 500 + 15_000)
```

Comment:
what are 500 and 15_000

Reply:
Added comments, `500ms` since we set ganache's` `blockTime` to mine every 0.5s and `15_000` was originally to account for delays of the network(reduce flakiness). I've turned that down to 5 seconds.